harden/shell_log_linux.go

```diff
@@ -0,0 +1,159 @@
+package harden
+
+import (
+	"fmt"
+	"os"
+	"os/exec"
+	"path/filepath"
+)
+
+func configureShell(filePath string, backupPath string, shellType string) {
+	absPath, err := filepath.Abs(filePath)
+	if err != nil {
+		fmt.Println("Error getting absolute path")
+		return
+	}
+
+	if _, err := os.Stat(absPath); os.IsNotExist(err) {
+		err = os.Mkdir(absPath, 0755)
+		if err != nil {
+			fmt.Println("Error creating history directory:", err)
+			return
+		}
+	}
+
+	var shellConfigFile string
+	if shellType == "bash" {
+		shellConfigFile = "/etc/bash.bashrc"
+	} else if shellType == "sh" {
+		shellConfigFile = "/etc/profile"
```

Comment:
Since you mention that sh logging messes with the shell, I think its best to drop sh support for now given the tight deadline.